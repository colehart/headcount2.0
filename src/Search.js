import React, { Component } from 'react'

class Search extends Component {
  render() {
    return(
      <form className="Search">
      </form>
    )
  }
}

export default Search