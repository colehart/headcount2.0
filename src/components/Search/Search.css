/*SEARCH COMPONENT RELATED CSS GOES HERE*/
.Search {
  grid-row: 2 / 3;
  grid-column: 1 / 2;
  width: 50%;
}

input {
  height: 2rem;
  width: 100%;
  border: 2px solid #9E9E9E;
  padding-left: .6rem;
  font-size: 1.25rem;
  letter-spacing: 0.5px;
}

::placeholder {
  font-family: "Roboto", sans-serif;
  font-weight: 400;
  color: #9E9E9E;
}