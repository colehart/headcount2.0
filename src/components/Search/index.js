import React, { Component } from 'react'
import PropTypes from 'prop-types'
import './Search.css'

class Search extends Component {
  constructor({findAllMatches}) {
    super()
    this.state = {
      query: ''
    }
  }

  handleInputChange = (event) => {
    const { name, value } = event.target

    this.setState({ [name]: value })
    this.submitQuery()
  }

  submitQuery = () => {
    this.props.findAllMatches(this.state.query);
  }

  handleSubmit = (event) => {
    event.preventDefault()
    this.handleInputChange(event)
  }

  render() {
    return(
      <form className="Search" onSubmit={this.handleSubmit} >
        <input
          className="search-input"
          aria-label="Search for a specific district here"
          placeholder="Search for a specific district"
          name="query"
          value={this.state.query}
          onChange={this.handleInputChange}
        />
      </form>
    )
  }
}

Search.propTypes = {
  findAllMatches: PropTypes.func.isRequired
}

export default Search