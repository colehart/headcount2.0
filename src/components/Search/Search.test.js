import React from 'react'
import { shallow } from 'enzyme'
import Search from './index'

describe('Search', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Search />)
  })

  it('should have a search input and a button', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should search for matching districts onChange and onSubmit', () => {

  })
})