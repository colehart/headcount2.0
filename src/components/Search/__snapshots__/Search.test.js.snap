// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should have a search input and a button 1`] = `
<form
  className="Search"
/>
`;
